devito/symbolics/queries.py

```diff
@@ -22,10 +22,14 @@
 """
 
 
-def q_scalar(expr):
+def q_number(expr):
```

Comment:
this name is misleading. Can we stick to `q_scalar`? 

as for the new one, maybe we could use `q_symbol`, then rename `retrieve_scalars`  as `retrieve_symbol`, and use `retrieve_symbol` in `collect_temp` ? after all, a Scalar is just a Symbol subclass. I however appreciate that there is some naming mess up

Reply:
Not sure I follow or agree, feels like semantics preferences

Reply:
it's convention.
1) SymPy calls Number objects that are non-symbolic numbers. 
2) Devito calls Scalars objects that are subclass of Symbol
so either we change Scalar all over the place (...) or we tweak this name

Reply:
`q_scalar` check if it is a `Scalar`, why the hell would I name this one like that that makes zero sense